src/components/Banners/YatBanner.tsx

```diff
@@ -16,8 +16,14 @@ type YatBannerProps = {
 
 const hoverProp = { boxShadow: '0 0 0 2px var(--chakra-colors-chakra-body-text) inset' }
 
-export const YatBanner: React.FC<YatBannerProps> = memo(({ isCompact, ...rest }) => {
-  const [isLargerThan2xl] = useMediaQuery(`(min-width: ${breakpoints['2xl']})`, { ssr: false })
+export const YatBanner: React.FC<YatBannerProps> = memo(props => {
+  const { isCompact, ...linkProps } = useMemo(() => {
```

Comment:
One concern here (I mean, not actually here here as this file has been removed) as this pattern is growing out across the app re: `const {foo, bar} = useMemo(() => (someExpression))` here - fine if we destructure primitives (i.e `isCompact`), not so fine if we destructure objects (i.e objects, arrays or functions), since destructuring effectively creates a new pointer every render, which could leak to potential re-renders.
i.e across renders:
`prevIsCompact === isCompact` but `prevLinkProps !== linkProps`

I could be wrong here and this can be verified with `usePrevious()` but just pointing this out so we double check this and don't spread this pattern if it is not needed

Reply:
you're 100% correct here ser, though this file no longer exists but will keep in mind for other instances